client/pages/sources/show/overview/map.js

```diff
@@ -61,36 +63,53 @@ function getColorForWheelchairAccessiblity(placeData) {
 function centerOnCurrentPlace(map) {
   const place = getCurrentPlaceInfo();
   if (place) {
-    map.setView(place.geometry.coordinates.reverse(), 16);
+    map.setView(place.geometry.coordinates.reverse(), 18);
   }
 }
 
-function showPlacesOnMap(map, geoMarkerData) {
-  const geojsonLayer = new L.geoJson(geoMarkerData, { // eslint-disable-line new-cap
-    pointToLayer(feature, latlng) {
-      const categoryIconName = _.get(feature, 'properties.category') || 'place';
-      const color = getColorForWheelchairAccessiblity(feature);
+const idsToShownMarkers = {};
 
-      const acIcon = new L.AccessibilityIcon({
-        iconUrl: `/icons/categories/${categoryIconName}.png`,
-        className: `ac-marker ${color}`,
-        // iconSize: [27, 27],
-      });
-      const marker = L.marker(latlng, { icon: acIcon });
-      marker.on('click', () => {
-        FlowRouter.go('placeInfos.show', {
-          _id: FlowRouter.getParam('_id'),
-          limit: FlowRouter.getParam('limit'),
-          placeInfoId: feature.properties._id,
-        });
-      });
+function filterShownMarkers(featureCollection) {
+  const result = {};
+  result.features = featureCollection.features
+    .filter(feature => !idsToShownMarkers[feature.properties._id]);
+  result.featureCount = result.features.length;
+  return result;
+}
 
-      return marker;
-    },
-  });
+function showPlacesOnMap(markerClusterGroup, map, unfilteredFeatureCollection) {
+  const featureCollection = filterShownMarkers(unfilteredFeatureCollection);
+
+  // console.log('Adding', featureCollection.features.length, 'new places to map...');
+  if (featureCollection.features && featureCollection.features.length) {
```

Comment:
Another opportunity for return early :)
NABD ("Not A Big Deal" or - feel free to address this or not, can be done at a later time)